localstack/deprecations.py

```diff
@@ -303,6 +303,22 @@ def log_deprecation_warnings(deprecations: Optional[List[EnvVarDeprecation]] = N
     affected_deprecations = collect_affected_deprecations(deprecations)
     log_env_warning(affected_deprecations)
 
+    provider_override_lambda = os.environ.get("PROVIDER_OVERRIDE_LAMBDA")
+    if provider_override_lambda and provider_override_lambda in ["v1", "legacy"]:
```

Comment:
Do we want/need this prominently at startup?

<img width="2840" alt="Screenshot 2023-09-21 at 20 43 08" src="https://github.com/localstack/localstack/assets/8161292/66d4c6ec-bc3a-4321-9037-ed0d0923ee9e">

Reply:
Yes. The Lambda team discussed this makes sense for Lambda given the substantial difference. We want to highlight this and push users to adopt v2 or raise any blocking issues.

Reply:
Notice that the language differs from the lazily loaded message. This message is aligned to the regular LS-startup deprecation warnings.